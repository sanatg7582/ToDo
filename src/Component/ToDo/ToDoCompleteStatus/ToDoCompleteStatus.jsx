import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faCheckCircle,
  faTimesCircle
} from "@fortawesome/free-solid-svg-icons";
import Button from "../../Shared/Button";

/**
 * This function return Is complete Button
 * @param {bool} isCompleteStatus,
 * @param {func} handleOnClick,
 * @param {number} index
 * @returns Element
 */
const ToDoCompleteStatus = props => {
  const { isCompleteStatus, handleOnClick, index } = props;
  return (
    <>
      {isCompleteStatus ? (
        <Button
          buttonType="button"
          btnClass="success"
          handleOnClick={handleOnClick}
          index={index}
          status={true}
        >
          <FontAwesomeIcon icon={faCheckCircle} />
        </Button>
      ) : (
        <Button
          buttonType="button"
          btnClass="danger"
          handleOnClick={handleOnClick}
          index={index}
          status={false}
        >
          <FontAwesomeIcon icon={faTimesCircle} />
        </Button>
      )}
    </>
  );
};
// default props value
ToDoCompleteStatus.defaultProps = {
  isCompleteStatus: false,
  handleOnClick: () => {},
  index: 0
};
// prop type validation
ToDoCompleteStatus.propTypes = {
  isCompleteStatus: PropTypes.bool.isRequired,
  handleOnClick: PropTypes.func.isRequired,
  index: PropTypes.number
};
export default ToDoCompleteStatus;
