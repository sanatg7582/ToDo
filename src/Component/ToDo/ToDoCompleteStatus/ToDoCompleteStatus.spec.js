import React from "react";
import { mount } from "enzyme";
import ToDoCompleteStatus from "./ToDoCompleteStatus";
describe("Load <ToDoCompleteStatus />", () => {
  let wrapper = "";
  beforeAll(() => {
    const props = {
      isCompleteStatus: false,
      handleOnClick: () => {},
      index: 0
    };
    wrapper = mount(<ToDoCompleteStatus {...props} />);
  });
  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy();
  });
  it("checking button length", () => {
    expect(wrapper.find("button")).toHaveLength(1);
  });
  it("checking props", () => {
    // todo status true
    expect(wrapper.find("button").hasClass("btn danger")).toEqual(true);
    expect(wrapper.props().isCompleteStatus).toEqual(false);

    // todo status wrong
    wrapper.setProps({ isCompleteStatus: true });
    expect(wrapper.find("button").hasClass("btn success")).toEqual(true);
    expect(wrapper.props().isCompleteStatus).toEqual(true);
  });
  it("Checking button click", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.props().isCompleteStatus).toEqual(true);
  });
  it("SnapShot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
