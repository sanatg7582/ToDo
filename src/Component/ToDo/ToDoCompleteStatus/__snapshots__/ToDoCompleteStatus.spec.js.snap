// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load <ToDoCompleteStatus /> SnapShot test 1`] = `ReactWrapper {}`;
